api/src/main/java/edu/cornell/mannlib/vitro/webapp/auth/objects/DataPropertyAccessObject.java

```diff
@@ -0,0 +1,52 @@
+package edu.cornell.mannlib.vitro.webapp.auth.objects;
+
+import edu.cornell.mannlib.vitro.webapp.auth.attributes.AccessObjectType;
+import edu.cornell.mannlib.vitro.webapp.beans.DataProperty;
+import edu.cornell.mannlib.vitro.webapp.web.templatemodels.individual.FauxDataPropertyWrapper;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class DataPropertyAccessObject extends AccessObject {
+
+    private static final Log log = LogFactory.getLog(DataPropertyAccessObject.class);
+
+    public DataPropertyAccessObject(DataProperty dataProperty) {
+        setDataProperty(dataProperty);
+        debug(dataProperty);
+    }
+
+    @Override
+    public String getUri() {
+        DataProperty dp = getDataProperty();
+        if (dp != null) {
+            return dp.getURI();
+        }
+        return null;
+    }
+
+    @Override
+    public AccessObjectType getType() {
+        return AccessObjectType.DATA_PROPERTY;
+    }
+
+    @Override
+    public String toString() {
+        DataProperty dp = getDataProperty();
+        return getClass().getSimpleName() + ": " + (dp == null ? dp : dp.getURI());
+    }
+
+    private void debug(DataProperty dataProperty) {
+        if (true) {
+            if (dataProperty instanceof FauxDataPropertyWrapper) {
+                Throwable t = new Throwable();
+                log.error("FauxDataPropertyWrapper provided in DataPropertyAccessObject constructor");
+                log.error(t, t);
+            }
+            if (dataProperty == null) {
+                Throwable t = new Throwable();
+                log.error("null provided in DataPropertyAccessObject constructor");
+                log.error(t, t);
+            }
+        }
+    }
+}
```

Comment:
no empty line at the end of the document

Reply:
fixed